### PR TITLE
Issue #14 Generate config key

### PR DESCRIPTION
**Pull request by andrewmadden**
Dependent on `property_list` to export JSON with SEB Config Key specifications.

### CODE REVIEW

File: tests/phpunit/config_key_test.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+class quizaccess_seb_config_key_testcase extends advanced_testcase {
+
+    /**
+     * Test that the Config Key hash is generated.
+     */
+    public function test_config_key_hash_generated() {
+        $xml = file_get_contents(__DIR__ . '/sample_data/unencrypted.seb');
+        $hash = config_key::generate($xml)->get_hash();
+        $this->assertEquals('9683df435f3a226d9d8b206846910ff1045c18b13e518474541f40461a61afee', $hash);
```
Comment (dmitriim) [inline tests/phpunit/config_key_test.php]: how did you get this hash?

File: classes/config_key.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+     * See  https://safeexambrowser.org/developer/seb-config-key.html for more information about the process.
+     *
+     * @param string $xml A PList XML string, representing SEB config.
+     * @return config_key This config key instance.
+     *
+     * @throws \CFPropertyList\IOException
+     * @throws \CFPropertyList\PListException
+     * @throws \DOMException
+     */
+    public static function generate(string $xml) {
```
Comment (dmitriim) [inline classes/config_key.php:56]: can we please add tests for broken XML? are we expecting exceptions?

**Review by dmitriim — CHANGES_REQUESTED** [review]

File: classes/config_key.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        $plist = new property_list($xml);
+        // Remove the key "originatorVersion" first. This key is exempted from the SEB-JSON hash (it's a special key
+        // which doesn't have any functionality, it's just meta data indicating which SEB version saved the config file).
+        $plist->delete_element('originatorVersion');
```
Comment (dmitriim) [inline classes/config_key.php:60]: what will happen if the key is not exist at all? error? we need to cover this by tests

File: tests/phpunit/config_key_test.php (hunk 1 above)
Comment (andrewmadden) [reply tests/phpunit/config_key_test.php]: This is tricky. 

At the moment this hash was created from the code it's testing, so not very good. It at least shows that the process hasn't changed.

I might need some help figuring out how to get some real sample data to test this against.

File: tests/phpunit/config_key_test.php [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    /**
+     * Test that the Config Key hash is generated.
+     *
+     * The file was generated using SEB 2.1.4 (Mac OS), and the expected Config Key extracted from the config tool.
+     */
+    public function test_config_key_hash_generated() {
+        $xml = file_get_contents(__DIR__ . '/sample_data/unencrypted_mac_001.seb');
+        $hash = config_key::generate($xml)->get_hash();
+        $this->assertEquals('4fa9af8ec8759eb7c680752ef4ee5eaf1a860628608fccae2715d519849f9292', $hash);
```
Comment (dmitriim) [inline tests/phpunit/config_key_test.php:50]: Can you please add comments how you generated this hash. Basically for debugging if something went wrong and we won't need to invent the wheel again.

Comment (andrewmadden) [reply tests/phpunit/config_key_test.php:50]: Good idea. Done.

**Pull request merged**